lib/mongoid/persistable/updatable.rb

```diff
@@ -186,6 +187,18 @@ def process_touch_option(options, children)
           children.each(&:timeless)
         end
       end
+
+      # Checks to see if the _id field has been modified. If it has, and if
+      # the document has already been persisted, this is an error. Otherwise,
+      # returns without side-effects.
+      #
+      # @raise [ Errors::ReadonlyAttribute ] if _id has changed, and document
+      #   has been persisted.
+      def enforce_immutability_of_id_field!
+        if _id_changed? && persisted?
+          raise Errors::ReadonlyAttribute.new(:_id, _id)
```

Comment:
`ReadonlyAttribute` error message contains the following wordings:
```yml
summary: "Attributes flagged as readonly via Model.attr_readonly
  can only have values set when the document is a new record."
resolution: "Don't define '%{name}' as readonly, or do not attempt
  to update its value after the document is persisted."
```
This is misleading in case of `_id` field, since it is not explicitly flagged as readonly. I suggest raising a dedicated error here.

Reply:
Yeah, good point. I've updated the PR with a new `ImmutableAttribute` exception.